sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategySuite.scala

```diff
@@ -487,6 +487,36 @@ class FileSourceStrategySuite extends QueryTest with SharedSQLContext with Predi
     }
   }
 
+  test("SPARK-19352: Keep sort order of rows after external sorter when writing") {
```

Comment:
again, this is not guaranteed, we should not test it.

This is an optimization and advanced users can leverage this to preserve the sort order, but it may change in the future.